### DISCUSSION TITLE
Auto Sign Out for Admin Users only

### DISCUSSION THREAD

**Discussion by ybelenko** [directus/directus#11714] [Feature Requests]
There was a setting in Directus 7 called `auto_sign_out`. How I can achieve the same behaviour in Directus 9?

**Comment by joselcvarela** [comment]
Hello,
You can make use of `ACCESS_TOKEN_TTL` to limit how long a session can be active. By default is `15m` (15 minutes). But it's possible to renovate a session with a refresh token.
The refresh token is the one who completely logs you out. To control the lifetime of this token you can change env var 
`REFRESH_TOKEN_TTL`. By default is `7d` which means 7 days

**Reply by ybelenko** [reply]
> Hello,
> You can make use of `ACCESS_TOKEN_TTL` to limit how long a session can be active. By default is `15m` (15 minutes). But it's possible to renovate a session with a refresh token.
> The refresh token is the one who completely logs you out. To control the lifetime of this token you can change env var
> `REFRESH_TOKEN_TTL`. By default is `7d` which means 7 days

I know about tokens TTL. I want to provide smooth experience for API users(`15m/7days`) and restricted for admin app user(logout after 30 mins of idle). Current Directus app refreshes token automatically, so even if he left his desktop for an hours dashboard will be open and visible.

It seems to me that it doesn't related to tokens because counting idle time(no key strokes, no mouse moves etc.) is responsibility for a front app javascript.

That `auto sign out` from admin app option was a request after security audit. So it's not even my personal need 😄

**Reply by joselcvarela** [reply]
I moved this to `Ideas` since it's not implemented.
In the meantime you could try to use custom hooks (`action('auth.login')`) and modifying `expires_in` in `directus_sessions` table to achieve what you intend:
https://github.com/directus/directus/blob/main/api/src/services/authentication.ts#L220

**Reply by ybelenko** [reply]
> In the meantime you could try to use custom hooks (`action('auth.login')`) and modifying `expires_in` in `directus_sessions` table to achieve what you intend

How is `directus_sessions` table useful? If I update all session records for admin user role to `now() + 30 mins` he will be signout in 30 minutes even if he wasn't idle. That's not what I need. Session should be long for admin user, but if he left his laptop for 30 mins then instant token revocation performed.

**Reply by joselcvarela** [reply]
This seems an edge case to me. Even services like AWS have a token which longs a full day.

**Reply by ybelenko** [reply]
I agree that it's an edge case. I just thought that Directus 7 auto sign out feature worked the way I described. I thought it's not connected to tokens, because tokens TTLs was already there as different config.

Anyway, maybe I can implement it myself and push PR then. If understand correct, all I need is to track user activity in the app Javascript and trigger `POST /auth/logout` to kill both access and refresh tokens when there is no browser activity for `n` minutes. Does `POST /auth/logout` call destroys refresh token too?

**Reply by benhaynes** [reply]
We have a feature coming out soon that could help with this. Allowing something like:

* Interval trigger
* Fetch users, filter by last login vs last activity
* Process a logout for anyone idle for x time

**Reply by Jeyakumarrathnaselvi** [reply]
Is this available already in the latest version? if yes how do we enable this?

**Reply by rijkvanzanten** [reply]
@Jeyakumarrathnaselvi No, see https://github.com/directus/directus/pull/12522 🙂

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
